sw/device/lib/testing/test_framework/ottf_isrs.S

```diff
@@ -19,11 +19,20 @@
    * b11 (0x3), which means that the trapped instruction is not compressed,
    * i.e., the trapped instruction is 32bits = 4bytes. Otherwise, the trapped
    * instruction is 16bits = 2bytes.
+   *
+   * Before loading the instruction pointed by the `mepc` we should check
+   * whether the address is invalid by checking the `mcause` register for the
+   * `Instruction access fault` code. It is invalid then return `mpec`.
```

Comment:
To confirm, this is the behavior you were seeing?

```suggestion
   * `Instruction access fault` code (1u). If it is invalid, return the unmodified `mpec`, as attempting to load an instruction at an invalid address below will cause another exception.
```

Reply:
Yes, exactly that!